12-offer-encoding.md

```diff
@@ -391,65 +407,88 @@ for [Signature Calculation](#signature-calculation).
-    - MUST set `invoice_request_quantity`
-    - MUST set it within that (inclusive) range.
-  - otherwise:
-    - MUST NOT set `invoice_request_quantity`
-  - if `offer_amount`:
-    - MUST specify `invoice_request_amount`.`msat` in multiples of the minimum lightning-payable unit
-      (e.g. milli-satoshis for bitcoin) for `chain` (or for bitcoin, if there is no `chain`).
-  - otherwise:
-    - MAY omit `invoice_request_amount`.
-    - if it sets `invoice_request_amount`:
-      - MUST specify `invoice_request_amount`.`msat` as greater or equal to amount expected by `offer_amount` (and, if present, `offer_currency`).
+  - if it sets `invoice_request_amount`:
+    - MUST set `msat` in multiples of the minimum lightning-payable unit
+        (e.g. milli-satoshis for bitcoin) for `invoice_request_chain` (or for bitcoin, if there is no `invoice_request_chain`).
   - if it supports bolt12 features:
     - MUST set `invoice_request_features`.`features` to the bitmap of features.
 
 The reader:
-  - MUST fail the request if `invoice_request_payer_key` is not present.
+  - MUST fail the request if `invoice_request_payer_id` is not present.
   - MUST fail the request if any fields have type greater or equal to 160.
   - if `invoice_request_features` contains unknown _odd_ bits that are non-zero:
     - MUST ignore the bit.
   - if `invoice_request_features` contains unknown _even_ bits that are non-zero:
     - MUST fail the request.
+  - if `offer_node_id` is present (response to an offer):
+    - MUST fail the request if the offer fields do not exactly match a valid, unexpired offer.
+    - MUST fail the request if `invoice_request_signature` is not correct as detailed in [Signature Calculation](#signature-calculation) using the `invoice_request_payer_id`.
+    - if `offer_quantity_min` or `offer_quantity_max` is present:
+      - MUST fail the request if there is no `invoice_request_quantity` field.
+      - MUST fail the request if `invoice_request_quantity` is not within that (inclusive) range.
+    - otherwise:
+      - MUST fail the request if there is an `invoice_request_quantity` field.
+    - if `offer_amount` is present:
+      - MUST calculate the *base invoice amount* using the `offer_amount`:
+        - if `offer_currency` is not the `invoice_request_chain` currency, convert to the
+          `invoice_request_chain` currency.
+        - if `invoice_request_quantity` is present, multiply by `invoice_request_quantity`.`quantity`.
+      - if `invoice_request_amount` is present:
+        - MUST fail the request if `invoice_request_amount`.`msat` is less than the *base invoice amount*.
+        - MAY fail the request if `invoice_request_amount`.`msat` exceeds the *base invoice amount*.
+        - MUST use `invoice_request_amount`.`msat` as the *base invoice amount*.
+    - otherwise (no `offer_amount`):
+      - MUST fail the request if it does not contain `invoice_request_amount`.
```

Comment:
This conflicts with the writer section:
```
    - if `offer_amount` is present:
      - MUST specify `invoice_request_amount`.
    - otherwise:
      - MAY omit `invoice_request_amount`.
```

Reply:
Good catch!  Writer section is backwards!  "if `offer_amount` is NOT present", i.e. if offer didn't indicate an amount, invoice request MUST.